test/functional/rpc_validateaddress.py

```diff
@@ -0,0 +1,136 @@
+        address = "BCRT1QPLMTZKC2XHARPPZDLNPAQL78RSHJ68U32RAH7R"
+        res = self.nodes[0].validateaddress(address, "bech32")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error_index'], 38)
+        assert_equal(res['error'], "Invalid")
+
+        # Valid multisig address
+        address = "bcrt1qdg3myrgvzw7ml9q0ejxhlkyxm7vl9r56yzkfgvzclrf4hkpx9yfqhpsuks"
+        res = self.nodes[0].validateaddress(address, "bech32")
+        assert_equal(res['isvalid'], True)
+
+        # Multisig address with 2 errors
+        address = "bcrt1qdg3myrgvzw7ml8q0ejxhlkyxn7vl9r56yzkfgvzclrf4hkpx9yfqhpsuks"
+        res = self.nodes[0].validateaddress(address, "bech32")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error_index'], 19)
+        assert_equal(res['error'], "Invalid")
+
+        # Address with 2 errors
+        address = "bcrt1qfsawymtaadjet5sl7p9zw273wpvu9rgwq93y52"
+        res = self.nodes[0].validateaddress(address, "bech32")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error_index'], 22)
+        assert_equal(res['error'], "Invalid")
+
+        # Invalid but also too long
+        address = "bcrt1q049edschfnwystcqnsvyfpj23mpsg3jcedq9xv049edschfnwystcqnsvyfpj23mpsg3jcedq9xv049edschfnwystcqnsvyfpj23m"
+        res = self.nodes[0].validateaddress(address, "bech32")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error_index'], 90)
+        assert_equal(res['error'], "String too long")
+
+        # Legacy address with no errors
+        address = "muehQnn2P5NiiRCGg9HewAWrXLRy2d7TZE"
+        res = self.nodes[0].validateaddress(address, "legacy")
+        assert_equal(res['isvalid'], True)
+
+        # Legacy address which is too long
+        address = "muehQnn2P5NiiRCGO9HewAWrXLRy2d7TZEaaaaaaa"
+        res = self.nodes[0].validateaddress(address, "legacy")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error'], "Length exceeds maximum for legacy and P2SH addresses")
+
+        # Legacy address with one error
+        address = "muehQnn2P5NhiRCGg9HewAWrXLRy2d7TZE"
+        res = self.nodes[0].validateaddress(address, "legacy")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error'], "Invalid checksum for Base58 address")
+        assert 'error_index' not in res
+
+        # Legacy address with a non-Base58 character
+        address = "muehQnn2P5NiiRCGO9HewAWrXLRy2d7TZE"
+        res = self.nodes[0].validateaddress(address, "legacy")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error'], "Invalid Base58 character in address")
+
+        # Legacy address with incorrect prefix for current network
+        address = "1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2"
+        res = self.nodes[0].validateaddress(address, "legacy")
+        assert_equal(res['isvalid'], False)
+        assert_equal(res['error'], "Invalid prefix for Base58 address")
```

Comment:
Same here - this address produces the error "Invalid Base58 character in address".